src/rag/vector_store/pgvector.py

```
@@ -1,28 +1,383 @@
+
+        with self._pool_lock:
+            # Double-check locking
+            if self._pool is not None:
+                return self._pool
+
+            try:
+                from psycopg_pool import ConnectionPool
+            except ImportError:
+                raise ImportError(
+                    "psycopg[pool] is required. Install with: pip install 'psycopg[binary,pool]'"
+                )
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)